android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt

```diff
@@ -49,22 +57,49 @@ open class QuestApplication : Application(), ConfigurableApplication {
   override val secureSharedPreference: SecureSharedPreference
     get() = SecureSharedPreference(applicationContext)
 
+  override val fhirPathEngine = FHIRPathEngine(workerContextProvider)
+
+  override val authenticatedUserInfo: UserInfo?
+    get() =
+      SharedPreferencesHelper.read(USER_INFO_SHARED_PREFERENCE_KEY, null)?.decodeJson<UserInfo>()
+
   override val resourceSyncParams: Map<ResourceType, Map<String, String>>
     get() {
-      return mapOf(
-        ResourceType.CarePlan to mapOf(),
-        ResourceType.Patient to mapOf(),
-        ResourceType.Questionnaire to mapOf(),
-        ResourceType.QuestionnaireResponse to mapOf(),
-        ResourceType.Binary to mapOf()
-      )
+      val searchParams = loadSearchParams(this)
+      val pairs = mutableListOf<Pair<ResourceType, Map<String, String>>>()
+      for (i in searchParams.indices) {
+        // TODO: expressionValue supports for Organization and Publisher, extend it using
+        // Composition resource
+        val expressionValue =
+          searchParams[i].expression?.let {
+            when {
+              it.contains("organization") -> authenticatedUserInfo?.organization
+              it.contains("publisher") -> authenticatedUserInfo?.questionnairePublisher
+              else -> null
+            }
+          }
+
+        pairs.add(
+          Pair(
+            ResourceType.fromCode(searchParams[i].base[0].code),
+            expressionValue?.let { mapOf(searchParams[i].expression to it) } ?: mapOf()
+          )
+        )
+      }
+
+      return mapOf(*pairs.toTypedArray())
     }
 
-  private fun buildPublisherFilterMap(): MutableMap<String, String> {
-    val questionnaireFilterMap: MutableMap<String, String> = HashMap()
-    val publisher = getPublisher()
-    if (publisher != null) questionnaireFilterMap[Questionnaire.SP_PUBLISHER] = publisher
-    return questionnaireFilterMap
+  private fun loadSearchParams(context: Context): List<SearchParameter> {
+    val iParser: IParser = FhirContext.forR4().newJsonParser()
+    val json = context.assets.open(SYNC_CONFIG).bufferedReader().use { it.readText() }
+    val searchParameters = mutableListOf<SearchParameter>()
+
+    val jsonArrayEntry = JSONArray(json)
+    for (i in 0 until jsonArrayEntry.length()) {
```

Comment:
is there a way to iterate through the contents directly instead of by index?

Reply:
I changed it to foreach, is it ok now?

Reply:
cool yea this looks like the best we can do because Kotlin doesn't have a JSON object iterator, which is what I was hoping for

Reply:
yeah, that's true.